src/test/frameworks/compose/discover/frameworkMatcher.spec.ts

```diff
@@ -0,0 +1,156 @@
+import { RepositoryFileSystem } from "../../../../frameworks/compose/discover/filesystem";
+import { expect } from "chai";
+import {
+  frameworkMatcher,
+  removeEmbededFrameworks,
+  filterFrameworksWithFiles,
+  filterFrameworksWithDependencies,
+} from "../../../../frameworks/compose/discover/frameworkMatcher";
+import { frameworkSpecs } from "../../../../frameworks/compose/discover/frameworkSpec";
+import { FrameworkSpec } from "../../../../frameworks/compose/discover/types";
+
+describe("frameworkMatcher", () => {
+  let fileSystem: RepositoryFileSystem;
+  const NODE_ID = "nodejs";
+
+  before(() => {
+    fileSystem = new RepositoryFileSystem("./src/frameworks/compose/discover/testapps/expressApp");
```

Comment:
Highly encourage we write a InMemory system to test out the matching algorithm... that's what Thomas intended IIUC.

Reply:
Changed to use `InMemory` tests.